From ca47ebec3bb1edf0c457e627bc2b359d9051a565 Mon Sep 17 00:00:00 2001
From: Flax <harlan@flax>
Date: Fri, 22 May 2026 20:03:35 +0000
Subject: [PATCH] Fix libpeci CPU-model enum names (skylake->skx, iceLake->icx)

This host-error-monitor revision predates the libpeci CPUModel rename;
peci.h defines skx/icx/icxd (not skylake/iceLake), so the LIBPECI IERR
decode path failed to compile. Map the two case labels in both files to
the current names. TiogaPass (Xeon Gold 6138) reports as skx.
---
 include/error_monitors/ierr_monitor.hpp | 4 ++--
 include/host_error_monitor.hpp          | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/error_monitors/ierr_monitor.hpp b/include/error_monitors/ierr_monitor.hpp
index b16ce17..41453c0 100644
--- a/include/error_monitors/ierr_monitor.hpp
+++ b/include/error_monitors/ierr_monitor.hpp
@@ -90,7 +90,7 @@ class IERRMonitor :
 
             switch (model)
             {
-                case skylake:
+                case skx:
                 {
                     // First check the MCA_ERR_SRC_LOG to see if this is the CPU
                     // that caused the IERR
@@ -184,7 +184,7 @@ class IERRMonitor :
                     }
                     break;
                 }
-                case iceLake:
+                case icx:
                 {
                     // First check the MCA_ERR_SRC_LOG to see if this is the CPU
                     // that caused the IERR
diff --git a/include/host_error_monitor.hpp b/include/host_error_monitor.hpp
index 4480256..7824bd0 100644
--- a/include/host_error_monitor.hpp
+++ b/include/host_error_monitor.hpp
@@ -199,7 +199,7 @@ static void printPECIError(const std::string& reg, const size_t addr,
 
         switch (model)
         {
-            case skylake:
+            case skx:
             {
                 // Check the ERRPINSTS to see if this is the CPU that
                 // caused the ERRx (B(0) D8 F0 offset 210h)
@@ -216,7 +216,7 @@ static void printPECIError(const std::string& reg, const size_t addr,
                 errPinCPUs[cpu] = (errpinsts & (1 << errPin)) != 0;
                 break;
             }
-            case iceLake:
+            case icx:
             {
                 // Check the ERRPINSTS to see if this is the CPU that
                 // caused the ERRx (B(30) D0 F3 offset 274h) (Note: Bus
-- 
2.34.1

